### DISCUSSION TITLE
Show only active route section - switch to "active" to late?

### DISCUSSION THREAD

**Discussion by WolfgangZ42** [devemux86/cruiser#534] [Q&A] [answered]
On my tour recently I learned that the switch from inactive to active route segment takes place just on the appropriate waypoint. As far as I remember, this was in the past done ~300m before reaching that point. Could you please check this? Thanks a lot!

**Comment by devemux86** [comment]
@WolfgangZ42 

It uses the "Small distance", which can be customized in settings, default for motor vehicles is 300 m.

**Reply by devemux86** [reply]
@WolfgangZ42 Have you changed the "Small distance" setting?

Or we should use a fixed distance for the change of the active route section?

**Reply by 0709wiwiwi** [reply]
The same thing was reported once before and was fixed then.  Now I haven't needed this for a while.

**Reply by WolfgangZ42** [reply]
No - Small distance is set to 300m, but I only marked Small distance in Navigation instructions. And all this is related to Voice guidance ... but the switch not. 
I'd prefer a fixed distance.

**Reply by devemux86** [reply]
This feature was implemented before the distances become adjustable,
so it inherited their changed value.

Indeed we can improve this with a fixed distance, as it is not related to voice guidance.

**Reply by bios4** [reply]
> It uses the "Small distance", which can be customized in settings, default for motor vehicles is 300 m.

Hmm...

My "Small distance" setting is 500 m, and the switch to the next active route section occurs FOR SURE later than these 500 meters!
I also have the feeling that it occurs only right before the appropriate waypoint!?

Running the latest 4.0.1-b3 by the way.

**Comment by devemux86** [comment]
There is indeed an issue when the next waypoint is a shaping point.
It will work properly again in the next Cruiser update.

Also we can change the distance of the change to the "Medium distance"?
(default for motor is 1.5 km)

**Reply by bios4** [reply]
Do you really want to connect it to a setting of voice guidance?
They are not related in principle...

**Reply by devemux86** [reply]
Such distances must be different per profile type: driving, cycling, walking.
So we could either use 3 fixed values or just use any existing distances.

**Reply by bios4** [reply]
I don't have concerns to re-use the voice guidance setting!

I just am not super happy with such "implicit relations" between non-related functions when there is no hint in the UI for the users...
Just my 0,05. 😉

**Reply by WolfgangZ42** [reply]
> Such distances must be different per profile type: driving, cycling, walking.

That's true!  👍 
And with this background I could live with the distances of the voice guidance.

**Comment by devemux86** [comment]
Show only active route section is improved in [Cruiser 4.0.1](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

**Reply by bios4** [reply]
Unfortunately I can NOT confirm that the behaviour has changed with 4.0.1!!!

The next route section is still only "activated" when a waypoint (shaping point in my case) is reached. Not even a little bit earlier... 😢

**Reply by devemux86** [reply]
@bios4 thanks for the report.

I just tested it again and it works.

Can you post a route sample?
Or a video to understand what you mean?
(can be with simulation and zoomed out)

**Reply by bios4** [reply]
Can't post a video because I'm on a multi-day trip in Slovenia, but I attach the first part of today's route (rename to .kurviger!)...

For none of the waypoints (well, shaping points) the earlier activation of the next route section worked, it was done only when reaching the shaping points.

[Lasko-Jurovci.txt](https://github.com/devemux86/cruiser/files/15375324/Lasko-Jurovci.txt)

**Reply by devemux86** [reply]
@bios4 thanks for the route.

Indeed it does not work with your route! I will check it.

**Reply by devemux86** [reply]
@bios4 please try [Cruiser 4.0.2-b1](https://github.com/devemux86/cruiser/discussions/522#discussioncomment-9496356).

**Reply by bios4** [reply]
Thanks @devemux86 - now it seems to work (tested in simulation mode)!

**Comment by devemux86** [comment, marked as answer]
Show only active route section is improved in [Cruiser 4.0.2](https://play.google.com/store/apps/details?id=gr.talent.cruiser).
